src/lib.rs

```diff
@@ -217,12 +212,37 @@ pub struct VirtualDestination {
 #[derive(PartialEq)]
 pub struct Device { object: Object }
 
+#[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
+type AlignmentMarker = [u32; 0]; // ensures 4-byte alignment (on ARM)
+
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+type AlignmentMarker = [u8; 0]; // unaligned
```

Comment:
What about other architectures than arm or x86 ? would it be possible to have the unaligned case for all the non arm-aarch64 architectures ?

Reply:
Yes, that would be possible. I'm just not sure wheter "unaligned" is the right default, so I did it in such a way that it would error on other architectures. Is macOS/iOS even available for other architectures?

Reply:
>> Is macOS/iOS even available for other architectures?
Good point. I guess that failing in case of unknown architecture is an acceptable solution here.